probe-rs/src/architecture/arm/ap/memory_ap/mock.rs

```diff
@@ -213,7 +213,7 @@ impl DapAccess for MockMemoryAp {
         }
     }
 
-    fn try_dap_probe(&self) -> Option<&dyn DapProbe> {
```

Comment:
I would appreciate if we had a method for both instead of changing all of them to `&mut` :)